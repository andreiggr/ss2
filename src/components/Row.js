import React from "react";

const Row = ({ row }) => {
  return (
    <div style={{display:"flex"}}>
      {row.map((item, i) => (
        <div
          style={{
            width: "150px",
            overflowWrap: "break-word",
            wordWrap: "break-word",
            border: "1px solid black",
            padding: "5px 5px 25px 5px"
          }}
          key={i}
        >
          <p>{item}</p>
        </div>
      ))}
    </div>
  );
};

export default Row;
