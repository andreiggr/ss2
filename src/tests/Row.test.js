import React from "react";
import ReactDOM from "react-dom";
import Row from "../components/Row";
import { shallow } from "enzyme";

describe("Row Render Test", () => {
  it("renders correctly the row with data", () => {
    const row = ["info1", "info2", "info3", "info4", "info5", "info6", "info7"];

    const appComponent = shallow(<Row row={row} />);
    expect(appComponent.getElements()).toMatchSnapshot();
  });
});
