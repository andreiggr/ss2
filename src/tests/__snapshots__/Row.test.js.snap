exports[`Row Render Test renders correctly the row with data 1`] = `
Array [
  <div
    style={
        Object {
            "display": "flex",
          }
    }>
    <div
        style={
            Object {
                "border": "1px solid black",
                "overflowWrap": "break-word",
                "padding": "5px 5px 25px 5px",
                "width": "150px",
                "wordWrap": "break-word",
              }
        }>
        <p>
            info1
        </p>
    </div>
    <div
        style={
            Object {
                "border": "1px solid black",
                "overflowWrap": "break-word",
                "padding": "5px 5px 25px 5px",
                "width": "150px",
                "wordWrap": "break-word",
              }
        }>
        <p>
            info2
        </p>
    </div>
    <div
        style={
            Object {
                "border": "1px solid black",
                "overflowWrap": "break-word",
                "padding": "5px 5px 25px 5px",
                "width": "150px",
                "wordWrap": "break-word",
              }
        }>
        <p>
            info3
        </p>
    </div>
    <div
        style={
            Object {
                "border": "1px solid black",
                "overflowWrap": "break-word",
                "padding": "5px 5px 25px 5px",
                "width": "150px",
                "wordWrap": "break-word",
              }
        }>
        <p>
            info4
        </p>
    </div>
    <div
        style={
            Object {
                "border": "1px solid black",
                "overflowWrap": "break-word",
                "padding": "5px 5px 25px 5px",
                "width": "150px",
                "wordWrap": "break-word",
              }
        }>
        <p>
            info5
        </p>
    </div>
    <div
        style={
            Object {
                "border": "1px solid black",
                "overflowWrap": "break-word",
                "padding": "5px 5px 25px 5px",
                "width": "150px",
                "wordWrap": "break-word",
              }
        }>
        <p>
            info6
        </p>
    </div>
    <div
        style={
            Object {
                "border": "1px solid black",
                "overflowWrap": "break-word",
                "padding": "5px 5px 25px 5px",
                "width": "150px",
                "wordWrap": "break-word",
              }
        }>
        <p>
            info7
        </p>
    </div>
</div>,
]
`;
